src/main/java/goblintserver/GoblintServer.java

```diff
@@ -0,0 +1,168 @@
+package goblintserver;
+
+import java.util.*;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Stream;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+import org.zeroturnaround.exec.InvalidExitValueException;
+import org.zeroturnaround.exec.ProcessExecutor;
+import org.zeroturnaround.exec.StartedProcess;
+
+import magpiebridge.core.MagpieServer;
+
+
+public class GoblintServer {
+
+    private MagpieServer magpieServer;
+
+    private File gobPieConf = new File("gobpie.json");
+    private File jsonResult = new File("analysisResults.json");
+    private File goblintSocket = new File("goblint.sock");
+
+    private String[] preAnalyzeCommand;
+    private String[] goblintRunCommand;
+
+    private final Logger log = LogManager.getLogger(GoblintClient.class);
+
+
+    public GoblintServer(MagpieServer magpieServer) {
+        this.magpieServer = magpieServer;
+    }
+
+
+    public String[] getPreAnalyzeCommand() {
+        return preAnalyzeCommand;
+    }
+
+
+    /**
+     * Method to start the Goblint server.
+     *
+     * @return True if server was started successfully, false otherwise.
+     */
+
+    public boolean startGoblitServer() {
+        // read configuration file
+        boolean gobpieconf = readGobPieConfiguration();
+        if (!gobpieconf) return false;
+
+        try {
+            // run command to start goblint
+            log.info("Goblint run with command: " + String.join(" ", goblintRunCommand));
+
+            StartedProcess commandRunProcess = runCommand(new File(System.getProperty("user.dir")), goblintRunCommand);
+
+            if (commandRunProcess.getFuture().isDone() && commandRunProcess.getProcess().exitValue() != 0) {
+                magpieServer.forwardMessageToClient(new MessageParams(MessageType.Error, "Goblint exited with an error."));
+                log.error("Goblint exited with an error.");
+                return false;
+            }
```

Comment:
This isn't very reliable in the long term: if Goblint crashes a bit later for any reason, it'd also be nice to know about it, not just when it immediately crashes.
A `ProcessListener` might work for that.